src/Common/tests/System/Net/Configuration.Certificates.cs

```diff
@@ -45,14 +45,24 @@ private static X509Certificate2Collection GetCertificateCollection(string certif
             {
                 // On Windows, .Net Core applications should not import PFX files in parallel to avoid a known system-level race condition.
                 // This bug results in corrupting the X509Certificate2 certificate state.
-                Assert.True(m.WaitOne(MutexTimeout), "Cannot acquire the global certificate mutex.");
-
-                var certCollection = new X509Certificate2Collection();
-                certCollection.Import(Path.Combine(TestDataFolder, certificateFileName), CertificatePassword, X509KeyStorageFlags.DefaultKeySet);
+                try
+                {
+                    Assert.True(m.WaitOne(MutexTimeout), "Cannot acquire the global certificate mutex.");
 
-                m.ReleaseMutex();
+                    var certCollection = new X509Certificate2Collection();
+                    certCollection.Import(Path.Combine(TestDataFolder, certificateFileName), CertificatePassword, X509KeyStorageFlags.DefaultKeySet);
```

Comment:
Thanks for clarifying! Then it might be worth investigating. 

I've already tested abrupt shutdown/crash: that will indeed cause the described issue. `Debug.Fail` seems to be the only particularity in this case.